collect_app/src/test/java/org/odk/collect/android/widgets/GeoPointWidgetTest.java

```diff
@@ -1,85 +1,301 @@
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.MAPS;
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.PLACEMENT_MAP;
+import static org.odk.collect.android.widgets.GeoPointWidget.ACCURACY_THRESHOLD;
+import static org.odk.collect.android.widgets.GeoPointWidget.DEFAULT_LOCATION_ACCURACY;
+import static org.odk.collect.android.widgets.GeoPointWidget.DRAGGABLE_ONLY;
+import static org.odk.collect.android.widgets.GeoPointWidget.LOCATION;
+import static org.odk.collect.android.widgets.GeoPointWidget.READ_ONLY;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.mockValueChangedListener;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.promptWithAnswer;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.promptWithAppearance;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.promptWithAppearanceAndAnswer;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.promptWithReadOnly;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.widgetTestActivity;
+import static org.robolectric.Shadows.shadowOf;
 
 /**
  * @author James Knight
  */
 
-@Config(shadows = {ShadowPlayServicesUtil.class})
-public class GeoPointWidgetTest extends BaseGeoWidgetTest<GeoPointWidget, GeoPointData> {
+@RunWith(RobolectricTestRunner.class)
+public class GeoPointWidgetTest {
 
-    @Mock
-    QuestionDef questionDef;
+    private final FakePermissionUtils permissionUtils = new FakePermissionUtils();
+    private final List<double[]> answerDoubles = getRandomDoubleArrayList();
+    private final String answer = stringFromDoubleList(answerDoubles);
 
-    private double[] initialDoubles;
-    private double[] answerDoubles;
+    private QuestionDef questionDef;
+    private MapConfigurator mapConfigurator;
+    private WaitingForDataRegistry waitingForDataRegistry;
 
-    @Override
-    public GeoPointData getInitialAnswer() {
-        return new GeoPointData(initialDoubles);
+    @Before
+    public void setup() {
+        questionDef = mock(QuestionDef.class);
+        mapConfigurator = mock(MapConfigurator.class);
+        waitingForDataRegistry = mock(WaitingForDataRegistry.class);
+        when(questionDef.getAdditionalAttribute(anyString(), anyString())).thenReturn(null);
     }
 
-    @NonNull
-    @Override
-    public GeoPointWidget createWidget() {
-        return new GeoPointWidget(activity, new QuestionDetails(formEntryPrompt, "formAnalyticsID"));
+    @Test
+    public void usingReadOnlyOption_makesAllClickableElementsDisabled() {
```

Comment:
Test and name don't match -- we decided the button should be gone, right? So the test name should reflect that.

Reply:
Right sorry! I will correct it.